data/src/main/java/com/sacramento/data/db/dao/MoviesDao.kt

```diff
@@ -1,24 +1,77 @@
 package com.sacramento.data.db.dao
 
-import androidx.room.Dao
-import androidx.room.Insert
-import androidx.room.OnConflictStrategy
-import androidx.room.Query
+import androidx.room.*
+import com.sacramento.data.db.models.MovieWithGenresDBModel
+import com.sacramento.data.db.tables.movie_tables.GenreEntity
+import com.sacramento.data.db.tables.movie_tables.MovieGenreCrossRef
 import com.sacramento.data.db.tables.movie_tables.SavedMovieEntity
-import kotlinx.coroutines.flow.Flow
 
 @Dao
 interface MoviesDao {
 
     @Insert(onConflict = OnConflictStrategy.REPLACE)
-    suspend fun insertSavedMovie(savedMovie: SavedMovieEntity)
+    suspend fun insertMovie(savedMovie: SavedMovieEntity)
 
-    @Query("DELETE  FROM saved_movie WHERE movieId IN (:movieId)")
-    suspend fun deleteSavedMovieById(movieId: List<Int>)
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    suspend fun insertGenre(genres: List<GenreEntity>)
+
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    suspend fun insertMovieGenreCrossRef(crossRef: List<MovieGenreCrossRef>?)
+
+    @Query("DELETE FROM genre")
+    suspend fun deleteFromGenre()
+
+    @Query("DELETE FROM saved_movie")
+    suspend fun deleteFromMoives()
+
+    @Query("DELETE FROM moviegenrecrossref")
+    suspend fun deleteFromCrossRef()
+
+    @Transaction
+    @Query(
+        "SELECT * FROM saved_movie JOIN moviegenrecrossref ON " +
+                "moviegenrecrossref.movieId = saved_movie.movieId WHERE " +
+                "moviegenrecrossref.genreId = :genreId "
+    )
+    suspend fun getMoviesByGenre(genreId: String?): List<MovieWithGenresDBModel>
```

Comment:
To prevent getting multilanguage results, try to add language parameter here like you do in getTopRatedMovies method.